proof-systems/src/darlin/tests/mod.rs

```diff
@@ -160,18 +178,20 @@ mod test {
             pcds[idx] = fake_pcds.unwrap()[idx].clone();
             vks[idx] = fake_vks.unwrap()[idx].clone();
 
-            let result = batch_verify_proofs::<G1, G2, D, R>(
+            let result = verify_aggregated_proofs::<G1, G2, D, R>(
                 pcds,
                 vks,
+                &proof_g1,
+                &proof_g2,
                 verifier_key_g1,
                 verifier_key_g2,
                 rng
             );
 
-            // Check AHP not failed in succinct part
-            assert!(!result.is_err() || result.unwrap_err().is_none());
+            // Check not failed in succinct part
+            assert!(!result.is_err() || result.clone().unwrap_err().is_none());
 
-            // Check AHP failed
+            // Check batch verification failed in hard part
```

Comment:
Check accumulation verification failed...

Reply:
Unlike batch_verify, this call can return Error too  and not only Ok(false). Please change the assertion

Reply:
I've changed the comment.

Regarding assertion: this assertion is already here:
assert!(!result.is_err() || result.clone().unwrap_err().is_none());

Isn't that what we need?

Reply:
Assertion on line 195 is one I have posted above:
assert!(!result.is_err() || result.clone().unwrap_err().is_none());

So it covers the case when error returned, not just Ok(true / false)

Sorry may be I missing something, but I cannot understand what wrong with it

Reply:
This is assertion at line 192 and it asserts it didn't fail in the succint part, and that's fine. But assertion at line 195, that asserts failure in hard part, shall be changed into assert!((result.is_err() && result.unwrap_err().is_none()) || (result.is_ok() && !result.unwrap()))

Reply:
This is assertion at line 192 and it asserts it didn't fail in the succint part, and that's fine. But assertion at line 195, that asserts failure in hard part, shall be changed into assert!((result.is_err() && result.unwrap_err().is_none()) || (result.is_ok() && !result.unwrap()))

Reply:
Actually both cases are covered now:

First (line 196):
assert!(!result.is_err() || result.clone().unwrap_err().is_none());

it covers the case Err(None)

Second (line 199)
assert!(!result.unwrap());

it covers the case (Ok(false))

Any way I have redone it as you have written above in a single line assertion